src/app/clusters/scenes/ExtensionFieldSetsImpl.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "ExtensionFieldSetsImpl.h"
+
+namespace chip {
+namespace scenes {
+
+// ExtensionFieldSetsImpl::ExtensionFieldSetsImpl() : ExtensionFieldSets() {}
+
+CHIP_ERROR ExtensionFieldSetsImpl::Serialize(TLV::TLVWriter & writer, TLV::Tag structTag) const
+{
+    TLV::TLVType container;
+    ReturnErrorOnFailure(writer.StartContainer(structTag, TLV::kTLVType_Structure, container));
+    // ReturnErrorOnFailure(writer.Put(TLV::ContextTag(TagEFS::kFieldSetsCount), static_cast<uint8_t>(mFieldSetsCount)));
+    ReturnErrorOnFailure(writer.StartContainer(TLV::ContextTag(TagEFS::kFieldSetArrayContainer), TLV::kTLVType_Array, container));
+    for (uint8_t i = 0; i < mFieldSetsCount; i++)
+    {
+        ReturnErrorOnFailure(mFieldSets[i].Serialize(writer));
+    }
+
+    return writer.EndContainer(container);
+    return writer.EndContainer(container);
+}
+
+CHIP_ERROR ExtensionFieldSetsImpl::Deserialize(TLV::TLVReader & reader, TLV::Tag structTag)
+{
+    TLV::TLVType container;
+    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, structTag));
+    ReturnErrorOnFailure(reader.EnterContainer(container));
+
+    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Array, TLV::ContextTag(TagEFS::kFieldSetArrayContainer)));
+    ReturnErrorOnFailure(reader.EnterContainer(container));
+
+    uint8_t i = 0;
+    CHIP_ERROR err;
+    while ((err = reader.Next(TLV::AnonymousTag())) == CHIP_NO_ERROR && i < kMaxClustersPerScene)
+    {
+        ReturnErrorOnFailure(mFieldSets[i].Deserialize(reader));
+        i++;
+    }
+    mFieldSetsCount = i;
+
+    VerifyOrReturnError(err == CHIP_END_OF_TLV, err);
+    return reader.ExitContainer(container);
```

Comment:
I guess maybe tests pass because this is also buggy and never exits the outer container?